core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -128,13 +130,10 @@ public String getName() {
 
     @Override
     protected Query doToQuery(QueryParseContext parseContext) throws IOException {
-        // make upstream queries ignore this query by returning `null`
-        // if either inner query builder is null or returns null-Query
-        if (positiveQuery == null || negativeQuery == null) {
-            return null;
-        }
         Query positive = positiveQuery.toQuery(parseContext);
         Query negative = negativeQuery.toQuery(parseContext);
+        // make upstream queries ignore this query by returning `null`
+        // if either inner query builder returns null-Query
```

Comment:
I'd prefer talking about `null` query here because we don't check for the new EmptyQueryBuilder here but any QueryBuilder returning `null` lucene query from `toQuery()`.